docs/devel/development.md

```diff
@@ -50,6 +50,14 @@ Official releases are built using Docker containers. To build Kubernetes using
 Docker please follow [these
 instructions](http://releases.k8s.io/HEAD/build/README.md).
 
+### Local OS/shell environment
+
+Many of the Kubernetes development helper scripts rely on a fairly up-to-date GNU tools
```

Comment:
Yeah, I considered that approach too.  But there are so many cases of this sort of thing (tools where we implicitly rely on some 'recentish', dont-quite-know-which-version tool, and I don't even know a reliable way to check the versions.  Can perhaps try in a follow-up PR, but lets get this doc update in in the mean time.